lib/db/mysql/index.js

```diff
@@ -28,67 +31,72 @@ function MysqlStore(options) {
   this._pool = mysql.createPool(options);
 }
 
-function createSchema(conn, options) {
-  logger.verbose('createSchema', options);
+// Apply patches up to the current patch level.
+// This will also create the DB if it is missing.
+
+function updateDbSchema(patcher) {
+  logger.verbose('updateDbSchema', patcher.options);
 
   var d = P.defer();
-  var database = options.database;
+  patcher.patch(function(err) {
+    if (err) {
+      logger.error('updateDbSchema', err);
+      return d.reject(err);
+    }
+    d.resolve();
+  });
+
+  return d.promise;
+}
 
-  logger.verbose('createDatabase');
-  var createDatabaseQuery =
-    'CREATE DATABASE IF NOT EXISTS ' + database +
-    ' CHARACTER SET utf8 COLLATE utf8_unicode_ci';
+// Sanity-check that we're working with a compatible patch level.
 
-  conn.query(createDatabaseQuery, function(err) {
+function checkDbPatchLevel(patcher) {
+  logger.verbose('checkDbPatchLevel', patcher.options);
+
+  var d = P.defer();
+
+  patcher.readDbPatchLevel(function(err) {
     if (err) {
-      logger.error('createDatabase', err);
+      logger.error('checkDbPatchLevel', err);
       return d.reject(err);
     }
-
-    logger.verbose('changeUser');
-    conn.changeUser({
-      user: options.user,
-      password: options.password,
-      database: database
-    }, function(err) {
-      if (err) {
-        logger.error('changeUser', err);
+    if (patcher.currentPatchLevel !== patch.level) {
+      if (patcher.currentPatchLevel !== patch.level + 1) {
```

Comment:
This tripped me for a second. I see now that if our patch level is 1 above the one in the DB, it's no problem because we'll be upgrading. Perhaps a comment to that effect?